app/pages/ServiceDiscoveryResults/ServiceDiscoveryResults.jsx

```diff
@@ -88,14 +88,35 @@ const InnerServiceDiscoveryResults = ({
   algoliaCategoryName,
   searchState,
   onSearchStateChange,
+  expandList,
+  setExpandList,
 }) => {
   const subcategoryNames = subcategories.map(c => c.name);
+  const searchResultsProps = { setExpandList, expandList };
 
   return (
     <div className={styles.container}>
       <div className={styles.header}>
         <h1 className={styles.title}>{categoryName}</h1>
+        <div className={styles.mapListToggler}>
+          <span
+            className={`${styles.listIcon} ${expandList ? styles.activeView : ''}`}
+            role="button"
+            tabIndex={0}
+            aria-label="list icon"
+            onClick={() => setExpandList(true)}
+          />
+          <span
+            className={`${styles.mapIcon} ${!expandList ? styles.activeView : ''}`}
+            role="button"
+            tabIndex={0}
+            aria-label="map icon"
+            onClick={() => setExpandList(false)}
+          />
```

Comment:
Good call on the button. I added a button parent class. I'm still using spans as the child element – I'm giving them a background-image prop and changing the background-position value depending on active state, since a single icon image file contains both the active and non-active image. If you think it'd be better to have the active/non-active images be separate files and to use an `img` tag w/dynamic src attribute, let me know and I'll make that change instead!

I like the idea of a segmented control component. If I get some downtime, I will work on that. It'd be useful. It would help to remove repeated code between the ServiceDiscoveryResults and the SearchResultsPage, for one thing

Reply:
Awesome, yeah, I think it's fine to continue using the image sprite approach. We may want to add back in some of the ARIA stuff, though, since then otherwise it may not be clear to screen readers or search engine crawlers what these buttons actually do.